flink-runtime/src/test/java/org/apache/flink/runtime/rest/RestServerEndpointITCase.java

```diff
@@ -501,10 +485,10 @@ public void testVersionSelection() throws Exception {
 
         try {
             version1Response.get(5, TimeUnit.SECONDS);
-            fail();
+            fail("Expected exception not thrown");
         } catch (ExecutionException ee) {
             RestClientException rce = (RestClientException) ee.getCause();
-            assertEquals(HttpResponseStatus.OK, rce.getHttpResponseStatus());
+            assertThat(rce.getHttpResponseStatus()).isEqualTo(HttpResponseStatus.OK);
         }
```

Comment:
I think we can use `assertThatFuture().eventuallyFailsWith(ExecutionException.class)...` as well.